src/test-rng.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * This file uses the Catch unit testing library, alongside
+ * testthat's simple bindings, to test a C++ function.
+ *
+ * For your own packages, ensure that your test files are
+ * placed within the `src/` folder, and that you include
+ * `LinkingTo: testthat` within your DESCRIPTION file.
+ */
+
+// All test files should include the <testthat.h>
+// header file.
+#include <testthat.h>
+#include "rng.h"
+#include "randlib.h"
+#include <Rcpp.h>
+
+using namespace Rcpp;
+
+void set_seed(double seed) {
+    Rcpp::Environment base_env("package:base");
+    Rcpp::Function set_seed_r = base_env["set.seed"];
+    set_seed_r(std::floor(std::fabs(seed)));
+}
+
+bool almost_equal(double a, double b, double delta) {
+	return a < b + delta && a > b - delta;
+}
+
+context("Random number generation") {
+  test_that("Exponential generation is reproducible") {
+	  set_seed(42);
+	  expect_true(almost_equal(rgenexp(1), 0.1983368, .0001));
```

Comment:
No harm in the way you've done it, but for future ref you can pass a tolerance argument in `testthat::expect_equal()`  to do this job a bit more succinctly

Reply:
Unfortunately, `expect_equal` doesn't seem to be available in c++ tests :(

Reply:
My bad! I just assumed everything from R testthat was available